flocker/dockerplugin/_api.py

```diff
@@ -157,3 +159,36 @@ def got_configured(configured):
         creating.addErrback(lambda reason: reason.trap(DatasetAlreadyExists))
         creating.addCallback(lambda _: {u"Err": None})
         return creating
+
+    @app.route("/VolumeDriver.Mount", methods=["POST"])
+    @_endpoint(u"Mount")
+    def volumedriver_mount(self, Name):
+        """
+        Move a volume with the given name to the current node and mount it.
+
+        Since we need to return the filesystem path we wait until the
+        dataset is mounted locally.
+
+        :param unicode Name: The name of the volume.
+
+        :return: Result indicating success.
+        """
+        dataset_id = UUID(dataset_id_from_name(Name))
+        d = self._flocker_client.move_dataset(self._node_id, dataset_id)
+
+        def get_state():
+            d = self._flocker_client.list_datasets_state()
+
+            def got_state(datasets):
+                datasets = [dataset for dataset in datasets
+                            if dataset.dataset_id == dataset_id]
+                if datasets and datasets[0].primary == self._node_id:
+                    return datasets[0].path
```

Comment:
What does it mean if there's more than one dataset with that id? I *think* if that happens then we actually want to fail hard because if we have multiple datasets with the same id we're not going to be able to do a good job with data integrity.

What do you think?